CourseGrab/ViewControllers/Search/SearchTableViewCell.swift

```diff
@@ -0,0 +1,137 @@
+    private let subtitleLabel = UILabel()
+    private let titleLabel = UILabel()
+    private let trackingButton = UIButton()
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        // Setup appearance
+
+        selectionStyle = .none
+
+        arrowImageView.backgroundColor = .black
+        arrowImageView.contentMode = .scaleAspectFit
+        containerView.addSubview(arrowImageView)
+
+        divider.backgroundColor = .courseGrabBorder
+        containerView.addSubview(divider)
+        divider.isHidden = true
+
+        containerView.clipsToBounds = false
+        containerView.layer.cornerRadius = 5
+        containerView.layer.shadowColor = UIColor.black.cgColor
+        containerView.layer.shadowOpacity = 0.3
+        containerView.layer.shadowRadius = 2
+        containerView.layer.shadowOffset = .zero
+        containerView.backgroundColor = .white
+        contentView.addSubview(containerView)
+
+        titleLabel.font = ._16Semibold
+        titleLabel.numberOfLines = 0
+        containerView.addSubview(titleLabel)
+
+        trackingButton.isHidden = true
+        trackingButton.layer.cornerRadius = 2
+        trackingButton.layer.borderWidth = 1
+        trackingButton.layer.borderColor = UIColor.courseGrabRuby.cgColor
+        trackingButton.titleLabel?.font = ._12Semibold
+        trackingButton.setTitle("REMOVE", for: .normal)
+        trackingButton.setTitleColor(.courseGrabRuby, for: .normal)
+        trackingButton.layer.borderColor = UIColor.courseGrabRuby.cgColor
+        containerView.addSubview(trackingButton)
+
+        statusBadge.contentMode = .scaleAspectFit
+        containerView.addSubview(statusBadge)
+        statusBadge.isHidden = true
+
+        subtitleLabel.font = ._14Semibold
+        containerView.addSubview(subtitleLabel)
+        subtitleLabel.isHidden = true
+
+        // Setup constraints
+
+        arrowImageView.snp.makeConstraints { make in
+            make.size.equalTo(16)
+            make.centerY.equalTo(titleLabel)
+            make.trailing.equalToSuperview().inset(16)
+        }
+
+        containerView.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview().inset(6)
+            make.leading.equalTo(20)
+            make.trailing.equalToSuperview().inset(20)
+        }
+
+        titleLabel.snp.makeConstraints { make in
+            make.trailing.equalTo(arrowImageView.snp.leading).offset(-8)
+            make.top.bottom.leading.equalToSuperview().inset(16)
+        }
+
+    }
+
+    func configure(for section: Section) {
+        titleLabel.text = section.title
+
+        if section.isTracking {
+            divider.isHidden = false
+            trackingButton.isHidden = false
+            statusBadge.isHidden = false
+            statusBadge.image = section.status.icon
+            subtitleLabel.isHidden = false
+            subtitleLabel.text = section.section
```

Comment:
Two things:
1. I'm assuming this means that if `isTracking` is false, we want `isHidden = true`? in this case we should move these before the if-block and do:
```
divider.isHidden = !section.isTracking
trackingButton.isHidden = !section.isTracking
...
```

2. If we're hiding the `statusBadge` and `subtitleLabel` for when `isTracking = true`, we can just always set their `image` and `text` without worrying about them

Reply:
Without the if-block, our cells would be the same height but without the additional information, which we don't want

![Simulator Screen Shot - iPhone 11 - 2020-02-11 at 13 53 16](https://user-images.githubusercontent.com/19507508/74268472-e6ac0680-4cd5-11ea-9819-97b02b817a98.png)

Reply:
Don't we need an `else` block to hide all those views in case the cell is reused for a section that isn't yet tracked?